File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        Queue<Integer> q = new LinkedList<>();
        
        for(int i=0; i<n; i++){
            if(visited[i]) {
                continue;
            }
            
            q.offer(i);
            visited[i] = true;
            answer++;
            
            while(!q.isEmpty()){
                int current = q.poll();
                
                for(int j=0; j<n; j++){
                    if(!visited[j] && computers[current][j] == 1){
                        q.offer(j);
                        visited[j] = true;
                    }
                }
            }
            
        }
        
        return answer;
    
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False] * n
    
    def dfs(v):
        visited[v] = True
        for nxt in range(n):
            if computers[v][nxt] == 1 and visited[nxt] == False:
                dfs(nxt)
    
    count = 0
    
    for i in range(n):
        if not visited[i]:
            dfs(i)
            count += 1
    
    return count